packages/realm-web/src/tests/utils/MockNetworkTransport.ts

```diff
@@ -47,12 +49,17 @@ export class MockNetworkTransport implements NetworkTransport {
     }
 
     /** @inheritdoc */
-    fetchAndParse<RequestBody extends any, ResponseBody extends any>(
+    fetch<RequestBody extends any>(
         request: Request<RequestBody>,
-    ): Promise<ResponseBody> {
+    ): Promise<FetchResponse> {
         if (!request.headers || Object.keys(request.headers).length === 0) {
             delete request.headers;
         }
+        // Save a parsed body, instead of a string
+        const { body } = request;
+        if (typeof body === "string") {
+            request.body = JSON.parse(body);
+        }
         if (!request.body) {
```

Comment:
That wouldn't be deleted? `'0'` is "truthy"? The empty string on the other hand, would be.

Reply:
`'0'` is truethy, but `JSON.parse('0')` is falsey

Reply:
Ah yeah, I didn't read your comment in context of the JSON.parse.

```suggestion
        if (typeof request.body === "object") {
```

Reply:
Why do this at all though? Wouldn't it be more likely to hide a bug? I think I'd delete the whole `if` block. Or maybe just do `if (request.body === undefined)` since I think the `Fetcher` will add an explicit `body: undefined`  which seems to be fine in practice, but confuses `deep.equals`

Reply:
Sure, let's do that. It seems more explicit.

```suggestion
        if (request.body === undefined) {
```